doc/Native APIs.md

```diff
@@ -0,0 +1,417 @@
+  child components which implement `student_view_user_state`. Dict keys are
+  component IDs and the value corresponding to each key is the output of
+  `student_view_user_state`(object). Child components which don't implement
+  `student_view_user_state` are not included in the dict.
+
+#### `student_results`
+
+The `student_results` field is an array of two-element arrays. Each nested array
+has the form of `[component_name, data]`. Only components which the student
+already submitted are present in the array. The structure of the `data` object
+depends on the type of the child component and is described separately for each
+type of child component below.
+
+### Custom Handlers
+
+#### `submit`
+
+The `submit` XBlock JSON handler is used for submitting student input. The
+`POST` data should be a dict where keys are child component names. The values
+depend on the component type and are described separately for each type of child
+component below.
+
+Returns a JSON object which contains these fields:
+
+- `results`: Same as `student_results` described under
+  `student_view_user_state` above.
+- `completed`: Same as `completed` described under `student_view_user_state`
+  above.
+- `message`: One of the `completed`, `incomplete`, or `max_attempts_reached`
+  feedback messages described under `messages` above. Can be `null` if no
+  messages has been defined. May contain HTML (string).
+- `max_attempts`: Same as `max_attempts` under `student_view_data`.
+- `num_attempts`: Same as `num_attempts` under `student_view_user_state`.
+
+
+Step Builder (`step-builder`)
+-----------------------------
+
+### `student_view_data`
+
+- `title`: The display name of the component (string).
+- `show_title`: `true` if the title should be displayed (boolean).
+- `weight`: The weight of the problem (float).
+- `extended_feedback`: `true` if extended feedback is enabled for this block
+  (boolean).
+- `max_attempts`: Max number of allowed attempts (integer).
+- `components`: A list of `student_view_data` output of all immediate child
+  components which implement `student_view_data`. For `step-builder`, immediate
+  children are typically `sb-step` and `sb-review-step` type components. Child
+  components which don't implement `student_view_data` are omitted from the list
+  (array).
+
+### `student_view_user_state`
+
+Same as Problem Builder above, but also contains these additional fields:
+
+- `active_step`: The index of the step which is currently active. Starts at zero
+  (integer).
+
+### Custom Handlers
+
+#### `submit`
+
+The `submit` XBlock JSON handler is used for submitting student input. The
+`POST` data should be a dict where keys are child component names. The values
+depend on the component type and are described separately for each type of child
+component below.
+
+In addition to child component names, the `POST` data should also include the
+`active_step` field. The value should be the index of the current step.
```

Comment:
Can you not submit answers to a step that is not currently active?  (Asking because I'm not sure)